src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
@vmg , I hadn't noticed this new *_with_mem API. I'm not sure I see the use case, but anyway the fact that "a library user should use the same allocator" should be documented, I couldn't find any mention of that in cmark.h :)

I figured out what the actual issue was here, and am not sure how to best solve it:

The "core extensions" are not statically linked with the executable, but dlopened, however we are passing -fvisibility=hidden to the compiler, and DEFAULT_MEM_ALLOCATOR ends up having a different address in cmark_node_new when called from the dlopened library. That's a pretty bad problem, especially as there is no API to retrieve the cmark_mem associated to a given parser, I can (and will) work around that in the core extensions because I #include private header files, including "parser.h", however externally defined extensions will not have that chance.

How do you think we should go about that?